solr/core/src/java/org/apache/solr/cloud/LeaderElector.java

```diff
@@ -131,14 +136,34 @@ private void checkIfIamLeader(final ElectionContext context, boolean replacement
     }
 
     if (leaderSeqNodeName.equals(seqs.get(0))) {
-      // I am the leader
+      // I am the leader - watch my node in case leadership is somehow revoked
+      try {
+        ElectionWatcher watcher = new ElectionWatcher(path, path, getSeq(path), context);
+        if (zkClient.exists(path, watcher, true) == null) {
```

Comment:
Yeah, that's fair enough about potentially leaking an exists watch (though I think the cost of that is low unless a single session creates a lot of them, and this is actually quite unlikely to happen).

The [recommended recipe](https://zookeeper.apache.org/doc/trunk/recipes.html#sc_recipes_Locks) suggests using an `exists` watch, but I'm guessing that's just to avoid downloading any data in the node. In this case, we're not storing anything in the node so there's nothing really wasteful about the `getData` call. So I'll change the original incantation back to the way it was and update this new one, too.